kinto/plugins/history/views.py

```diff
@@ -43,3 +45,88 @@ def _extract_filters(self, queryparams=None):
             filters_str_id.append(filt)
 
         return filters_str_id
+
+
+class Revert(resource.UserResource):
+
+    def __init__(self, request):
+        self.request = request
+        self.bucket_id = request.matchdict['bucket_id']
+        if 'since' not in request.params:
+            raise_invalid(self.request, 'params', 'since', 'since request parameter must be provided')
+        self.since = int(request.params['since'])
+        filters = super(Revert, self)._extract_filters(request.params)
+        filters = [s for s in filters
+                                 if s.field != 'since']
+        self.storage = request.registry.storage
+        self.history, _ = self.storage.get_all(parent_id='/buckets/%s' % self.bucket_id,
+                                               collection_id='history', filters=filters,
+                                               sorting=[Sort('last_modified', -1)])
+
+    def batch_revert(self, revert_list):
+        batch = {'path': '/batch', 'method': 'POST', 'body': {'requests': revert_list}}
```

Comment:
I think the code could be greatly simplified if this method would be in charge of building the requests from a list of data structures like tuples or dicts. This would prevent to have those `self.create_revert_operation()` everywhere